gnovm/pkg/gnolang/stacktrace.go

```diff
@@ -0,0 +1,79 @@
+package gnolang
+
+import (
+	"fmt"
+	"strings"
+)
+
+const maxStacktraceSize = 128
+
+type Execution struct {
+	Stmt  Stmt
+	Frame *Frame
+}
+type Stacktrace struct {
+	Executions []Execution
+}
+
+func (s Stacktrace) String() string {
+	var builder strings.Builder
+
+	excecutions := s.Executions
+	numFramesElided := 0
+	if len(s.Executions) > maxStacktraceSize {
+		excecutions = s.Executions[:maxStacktraceSize/2]
```

Comment:
Two questions here:
1. Why divide by 2 rather than set it equal to `s.Executions[:maxStacktraceSize]`?
2. If the output is limited by `maxStacktraceSize`, why bother building the `Stacktrace` object with values that won't be displayed? Perhaps the enforcing of the limit should be moved to where the object is built.

Reply:
1. I aim to make the stacktrace resemble Go more; hence, my decision.
2. I had no firm stance on the matter; thus, I will enforce the limit during struct creation.

Reply:
2. Done here: https://github.com/gnolang/gno/pull/2145/commits/248937512cfef4c11bc01c3b5ebd97af22d324ca

Reply:
> I aim to make the stacktrace resemble Go more; hence, my decision.
Can you link to what you're referencing? I don't understand why dividing by two is better than using a constant.

Reply:
If you run this example in https://go.dev/play/:

```go
package main

func p(i int) {
	if i == 200 {
		panic("here")
	}
	p(i + 1)
}

func main() {
	p(0)
}
```

You will notice that we have the first half of maxStacktraceSize of the stack trace followed by the last half  of maxStacktraceSize of the stack trace .

Reply:
Okay I see. Thanks for the clarification 👍